# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1319603017 25200
#      Tue Oct 25 21:23:37 2011 -0700
# Node ID ab4238b0b09b9ffa0a5b9dfa8c7f5ba9a482a19e
# Parent  6f068e3f3f9cba429fb3c82204a31a132b2350ed
Change return value of mxGetString to match Matlab (Bug #34546)

* mex.cc: Change return value to be 0 on success to match Matlab.

diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -1269,20 +1269,27 @@ public:
   void *get_imag_data (void) const { return pi; }
 
   void set_data (void *pr_arg) { pr = pr_arg; }
 
   void set_imag_data (void *pi_arg) { pi = pi_arg; }
 
   int get_string (char *buf, mwSize buflen) const
   {
-    int retval = 1;
+    int retval = 0;
 
     mwSize nel = get_number_of_elements ();
 
+    if (! (nel < buflen))
+      {
+        retval = 1;
+        if (buflen > 0)
+          nel = buflen-1;
+      }
+
     if (nel < buflen)
       {
         mxChar *ptr = static_cast<mxChar *> (pr);
 
         for (mwIndex i = 0; i < nel; i++)
           buf[i] = static_cast<char> (ptr[i]);
 
         buf[nel] = 0;
